vignettes/epipredict.Rmd

```diff
@@ -0,0 +1,360 @@
+---
+title: "Get started with epipredict"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{epipredict}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  echo = TRUE,
+  collapse = TRUE,
+  comment = "#>",
+  out.width = "100%")
+```
+
+```{r setup, message=FALSE}
+library(dplyr)
+library(parsnip)
+library(workflows)
+library(recipes)
+library(epiprocess)
+# remotes::install_github("cmu-delphi/epipredict")
+library(epipredict)
+```
+
+
+# Goals for the package
+
+At a high level, our goal with `epipredict` is to make running simple Machine Learning / Statistical forecasters for epidemiology easy. However, this package is extremely extensible, and that is part of the utility. Our hope is that it is easy for users with epi training and some statistics to fit baseline models while still allowing those with more nuanced statistical understanding to create complicated specializations within the same framework.
+
+Serving both populations is the main motivation for our efforts, but at the same time, we have tried hard to make it useful.
+
+
+## Baseline models
+
+We provide a set of basic, easy-to-use forecasters that work out of the box. 
+You should be able to do a reasonably limited amount of customization on them. Any serious customization happens with the framework discussed below).
+
+For the basic forecasters, we provide, at least: 
+    
+* Baseline flat-line forecaster 
+* Autoregressive forecaster
+* Autoregressive classifier
+
+All the forcasters we provide are built on our framework. So we will use these basic models to illustrate its flexibility.
+
+## Forecasting framework
+
+Our framework for creating custom forecasters views the prediction task as a set of modular components. There are four types of components:
+    
+1. Preprocessor: do things to the data before model training
+2. Trainer: train a model on data, resulting in a fitted model object
+3. Predictor: make predictions, using a fitted model object and processed test data
+4. Postprocessor: do things to the predictions before returning
```

Comment:
Nitpick but could we use language a bit more precise than "do things"? Maybe "manipulate data/predictions"?

Reply:
Oh, fine...